File: ParkMobileServer/Entities/Orders/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ParkMobileServer.Entities.Orders
{
    public enum OrderPayment
    {
        Card,
        QrCode,
        Cash,
        Credit
    }
    public enum OrderState
    {
        approved,
        disapproved,
    }
	public class Order
	{
        [Key]
        public int Id { get; set; }
        public string? Address {get; set;} = string.Empty;
        public string? PvzCode {get;set;} = string.Empty;
        public string? TrackNumber { get; set; } = null;

        public OrderPayment? Payment { get; set; } = null;
        public OrderState? State { get; set; } = null!;
        public int ClientId {get;set;}
        public OrderClient? Client { get; set; }
        public ICollection<OrderItem> Items {get; set;} = [];
    }
}

File: ParkMobileServer/park-mobile-ssr/src/Types/Order.ts
import { ReactNode } from "react"

export const PaymentDictionary = new Map([
  [0, "Карта"],
  [1, "QR-код"],
  [2, "Наличные"],
  [3, "Кредит"]
])

export enum OrderPayment
{
  Card = 0,
  QrCode,
  Cash,
  Credit
}
export enum OrderState
{
  Approved = 0,
  Disapproved,
}

export type Order = {
  id?: number
  address: string,
  pvzCode: string,
  payment: OrderPayment,
  state: OrderState | null,
  client: OrderClient, 
  items?: OrderItem[],
}

export type OrderClient = {
  clientName?: string,
  telephone?: string,
  email?: string,
  comment?: string,
}

export type OrderItem = {
  itemId: number,
  count: number
}

export type OrderTableData = Omit<Order, "client" | "items"> & { buttonBlock: ReactNode };
